src/DetectSecretsLinter.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright 2020 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Detect secrets in code base. 
+ */
+final class DetectSecretsLinter extends ArcanistExternalLinter {
+
+  public function getInfoName() {
+      return 'detect-secrets'; 
+  }
+
+  public function getInfoURI() {
+      return 'https://github.com/Yelp/detect-secrets';
+  }
+
+  public function getInfoDescription() {
+      return 'Detect secrets within code base during compile time';
+  }
+
+  public function getLinterName() {
+      return 'detect-secrets';
+  }
+
+  public function getLinterConfigurationName() {
+      return 'detect-secrets';
+  }
+
+  public function getDefaultBinary() {
+	  return 'detect-secrets';
+  }
+
+  public function getInstallInstructions() {
+      return pht('pip3 install detect-secrets');
+  }
+
+  protected function getMandatoryFlags() {
+      return array('-v', 'scan');
+  }
+
+  protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+
+      $lines = phutil_split_lines($stdout, false);
+
+      $messages = array();
+      $linter_results = array();
+
+      foreach($lines as $line) {
+         if ($line !== '') {
+            $linter_results[] = $line;
+         }
+      }
+
+      $results_keyword = "results";
+      $version_keyword = "version";
```

Comment:
Because these are constants, it would be better to include them directly in the regular expression patterns rather than use expensive string interpolation (repeated for in each loop iteration).